calculator/interfaces/Main.java

```diff
@@ -0,0 +1,141 @@
+                System.out.println("Choose your time zone");
+
+                for (Zone id : Zone.values()) {
+                    System.out.println(id.number + ": " + id);
+                }
+                int city = scan.nextInt();
+                scan.nextLine();
+                Zone d = Zone.PARIS;
+                for (Zone id : Zone.values()) {
+                    if (id.number == city) {
+                        System.out.printf("You choosed %s zone %n", id.idZone);
+                        d = id;
+                        break;
+                    }
+                }
+                calculatorDateAndTime = new CalculatorDateAndTime(d);
+                pattern = Pattern.compile("^([\\+\\-])\\s*((\\d*)y)?\\s*((\\d*)m)?\\s*((\\d*)d)?\\s*((\\d*)h)?\\s*((\\d*)min)?\\s*((\\d*)s)?$");
+                break;
+            }
+            else {
+                System.out.println("Bad command");
+            }
+        }
+        while (true) {
+            if (calculatorDateAndTime != null) {
+                System.out.println(calculatorDateAndTime.toString());
+                String input = StringHelper.removeWhitespace(scan.nextLine());
+                Matcher matcher = pattern.matcher(input);
+                Pattern patternForShowSince = Pattern.compile("^showSecondsSince(\\d{1,4})\\s?([1-9]|1[0-2])?\\s?([1-9]|1[0-9]|2[0-9]|3[01])?$"); 
+                Matcher matcherForShowSince = patternForShowSince.matcher(input);            
+                if (input.equals("exit")) {
+                    break;  
+                }
+                else if (matcher.matches()) {
+                    calculatorDateAndTime.calculate(input);
+                }
+                else if (input.equals("changeCity")) {
+                    for (Zone id : Zone.values()) {
+                        System.out.println(id.number + " " + id);
+                    }
+                    System.out.println("Enter the city number");
+                    while (true) {
+                        int city = scan.nextInt();
+                        scan.nextLine();
+                        if (city >= 1 && city <= 8) {
+                            for (Zone id : Zone.values()) {
+                                if (id.number == city) {
+                                    calculatorDateAndTime = new CalculatorDateAndTime(id);
+                                    break;
+                                }
+                            }
+                            break;       
+                        }
+                        else {
+                            System.out.println("Bad command enter correct number");
+                        }
+                    }
+                }
+                else if (input.equals("whenSunday")) {
+                    System.out.println(calculatorDateAndTime.whenSunday());
+
+                }
+                else if (matcherForShowSince.matches()) {
```

Comment:
not bad name, but type (Matcher) is added at end of the variable so showSecondsSinceMatcher